### ISSUE TITLE
Parameter propagation from Parameters dialog broken in OMEdit 1.22.0

### ISSUE THREAD

**Issue by pierre-haessig** [OpenModelica/OpenModelica#11695]
### Description

In OMEdit 1.22.0, doing **parameter propagation** from the GUI (Component Parameters dialog, see screenshot) is broken.

What I call "parameter propagation": in a toplevel system with: 

1. one component `comp` with its parameter `comp.x`  (*in the example code below, the component is `capacitor` with its parameter `capacitor.C`*)
2. one toplevel parameter `p` (*in the example code below, the toplevel parameter is `Ccircuit`*)

Symptom: it's not possible to set `comp.x=p` from the Component Parameters dialog of `comp`. That is with the model below, in the capacitor's Parameters GUI, it's impossible to set `capacitor.C` to `Ccircuit`.

### Steps to Reproduce

1. Load the following parameter in OMEdit 1.22.0:

```modelica
model ParameterPropagation "Bug in parameter propagation in OM 1.22.0"
  parameter Real Ccircuit;
  Modelica.Electrical.Analog.Basic.Capacitor capacitor(C = 3)  annotation(
    Placement(transformation(origin = {8, 14}, extent = {{-10, -10}, {10, 10}})));
equation

annotation(
    uses(Modelica(version = "4.0.0")));
end ParameterPropagation;
```

2. Open the capacitor's Parameters GUI and try setting its capacitance parameter `C` to `Ccircuit` 

![ParameterPropagation](https://github.com/OpenModelica/OpenModelica/assets/1010404/5a7ac898-631d-4c77-9d77-db59821c08a0)

### Expected Behavior

This GUI edit should change the code to `capacitor(C = Ccircuit) `, but no change happen.

Presently, the only way to achieve this is to edit the Modelica code.

### Version and OS

 - OpenModelica Version: OpenModelica 1.22.0
 - OS: tested in both Windows 10 and Ubuntu 22.04

**Comment by adeas31** [comment]
This looks similar to #11595 which is already fixed. The fix will be available in 1.22.1 release.

**Comment by pierre-haessig** [comment]
OK. The symptom in https://github.com/OpenModelica/OpenModelica/issues/11595 seems a bit more restrictive ("bug *(if the name is the same)*"), but perhaps the fix fixes everything.

Looking forward for the release of 1.22.1 then!
